contracts/token/FeiTimedMinter.sol

```diff
@@ -0,0 +1,104 @@
+pragma solidity ^0.8.0;
+
+import "../refs/CoreRef.sol";
+import "../utils/Timed.sol";
+import "../utils/Incentivized.sol";
+import "../utils/RateLimitedMinter.sol";
+import "./IFeiTimedMinter.sol";
+
+/// @title FeiTimedMinter
+/// @notice a contract which mints FEI to a target address on a timed cadence
+contract FeiTimedMinter is IFeiTimedMinter, CoreRef, Timed, Incentivized, RateLimitedMinter {
+    
+    /// @notice most frequent that mints can happen
+    uint256 public constant override MIN_MINT_FREQUENCY = 1 hours; // Min 1 hour per mint
+
+    /// @notice least frequent that mints can happen
+    uint256 public constant override MAX_MINT_FREQUENCY = 30 days; // Max 1 month per mint
+    
+    uint256 private _mintAmount;
+
+    /// @notice the target receiving minted FEI
+    address public override target;
+
+    /**
+        @notice constructor for FeiTimedMinter
+        @param _core the Core address to reference
+        @param _target the target for minted FEI
+        @param _incentive the incentive amount for calling mint paid in FEI
+        @param _frequency the frequency minting happens
+        @param _initialMintAmount the initial FEI amount to mint
+    */
+    constructor(
+        address _core,
+        address _target,
+        uint256 _incentive,
+        uint256 _frequency,
+        uint256 _initialMintAmount
+    ) 
+        CoreRef(_core)
+        Timed(_frequency)
+        Incentivized(_incentive)
+        RateLimitedMinter((_initialMintAmount + _incentive) / _frequency, (_initialMintAmount + _incentive), true) 
+    {
+        _initTimed();
+
+        _setTarget(_target);
+        _setMintAmount(_initialMintAmount);
+    }
+
+    /// @notice triggers a minting of FEI
+    /// @dev timed and incentivized
+    function mint() public virtual override whenNotPaused afterTime {
+
+        /// Reset the timer
+        _initTimed();
+
+        uint256 amount = mintAmount();
+
+        // incentivizing before minting so if there is a partial mint it goes to target not caller
+        _incentivize();
+
+        _mintFei(target, amount);
```

Comment:
Would be good to get some more comments overall saying that this file calls the rateLimitedMinter's over ridden `_mintFei` function which checks to ensure that amount of tokens can be minted according to the buffer.

Reply:
Added